pkg/plugin/conntrack/_cprog/conntrack.c

```diff
@@ -243,8 +246,10 @@ static __always_inline bool _ct_should_report_packet(struct ct_entry *entry, __u
 
     // Check if the connection timed out or if it is a TCP connection and FIN or RST flags are set.
     if (now >= eviction_time || (protocol == IPPROTO_TCP && flags & (TCP_FIN | TCP_RST))) {
-        // The connection is closing or closed. Mark the connection as closing. Update the flags seen and last report time.
-        WRITE_ONCE(entry->is_closing, true);
+        // The connection is closing or closed. Delete the connection from the map
+        bpf_map_delete_elem(&retina_conntrack, key);
+
+        // Update the flags seen and last report time.
```

Comment:
we don't have to update these if we deleted the entry, we could return true immediately